src/audio/copier/copier.c

```diff
@@ -740,6 +740,29 @@ static struct comp_dev *copier_new(const struct comp_driver *drv,
 	}
 
 	dev->pipeline = ipc_pipe->pipeline;
+	/* Calculation of the period in which the pipeline should be scheduled based on
+	 * the input buffer size.
+	 */
+	if (copier->gtw_cfg.dma_buffer_size > 0) {
+		/* Size of the single sample in bytes. */
+		uint32_t sample_size = copier->base.audio_fmt.channels_count *
+				       (copier->base.audio_fmt.depth / 8);
```

Comment:
doesnt this depend on direction? And maybe even restrict this to the host gateway case?

Reply:
No, right now we aim to use it on simple pipelines. Only when we have two connected gtw's.